### PR TITLE
Disallow compound assignment for tuples.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/1720

### CODE REVIEW

File: libsolidity/analysis/TypeChecker.cpp [hunk 1]
```diff
@@ -937,6 +937,11 @@ bool TypeChecker::visit(Assignment const& _assignment)
 	_assignment.annotation().type = t;
 	if (TupleType const* tupleType = dynamic_cast<TupleType const*>(t.get()))
 	{
+		if (_assignment.assignmentOperator() != Token::Assign)
```
Comment (yann300) [inline libsolidity/analysis/TypeChecker.cpp:950]: more `==` ?

Comment (chriseth) [reply libsolidity/analysis/TypeChecker.cpp:950]: No, it should be correct. please take a look at https://github.com/ethereum/solidity/pull/1749/files/77fd76cc2c06b33625d2dc1e6b602addf1033a08#diff-9f2a45efaa7b75ab7ec620d3177cf01aR954

**Review by pirapira — APPROVED** [review]
Looks good to me.

**Comment by pirapira** [comment]
@chriseth Please rebase

**Comment by chriseth** [comment]
@yann300 any remarks?

**Comment by yann300** [comment]
it's totally fine

**Pull request merged**